File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int i, j;
        string first = "", mid = "";
        vector<int> freq(26, 0);
        for (auto x : s) {
            freq[x - 'a']++;
        }
        for (i = 0; i < 26; i++) {
            first.append(freq[i] / 2, char('a' + i));
            if (freq[i] % 2 == 1) {
                mid.append(1, char('a' + i));
            }
        }
        string last = first;
        reverse(last.begin(), last.end());
        return first + mid + last;
    }
};

File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.py
class Solution(object):
    def smallestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        freq=26*[0]
        for c in s:
            freq[ord(c)-ord('a')]+=1
        left=""
        mid=""
        for i in range(len(freq)):
            left+= chr(ord('a') + i)*(freq[i]//2)
            if(freq[i]%2==1):
                mid+=chr(ord('a') + i)    
        right=left[::-1]
        return left+mid+right
        